### PR TITLE
DOCS Update documentation for SL installation and contribution, fix some typos

### PR DESCRIPTION
**Pull request by robbieaverill**
Notable changes:

* Removed suggestion for installing without composer
* Replace suggestion for putting GitHub issue numbers into commits, prefer putting them in PRs instead (so squashing and re-pushing doesn't clutter up an issue)
* Removed suggestion for using Git Flow, prefer PRs against major version branch you're using

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
-e.g. if you want to implement your awesome feature which can be summarised as "My Awesome Annotation", create a branch named "pulls/my-awesome-annotation".
-If you want to fix an issue mentioned in issues, create a branch named "pulls/issue-{#}-description-of-issue". Where {#} should be replaced with the issue number ofcourse.
+When raising a pull request, please raise it against the branch that represents the major version you're using. For example, if you're using `2.0.7` then you'll raise a pull request against the `2` branch. If you're using the latest unstable version, then raise it against the master branch.
 
-In case of the latter, please tag your commit with the issue number, by simply starting with `issue #15`, followed by a description of the issue that's fixed. This will tell github, your commit is linked to that specific issue.
+Please ensure that your commit messages are short and describe what you are doing. Do not include GitHub issue numbers in the commit messages, but please ensure that you enter them into the pull request description.
```
Comment (Firesphere) [inline CONTRIBUTING.md]: Commit messages shouldn't be short, they should be descriptive.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
 
-Please try to keep everything in one commit. If needed, squash and force-push.
+Please try to keep everything in one commit. If needed, squash and force-push (carefully).
 
-Rules above do not apply for collaborators/owner, who work on feature branches and/or hotfixes directly. (We have awesomeness powers)
+## Code style
 
-#Code style
+Please follow the [PSR-2 coding style guide](http://www.php-fig.org/psr/psr-2/) when editing or adding new code.
```
Comment (Firesphere) [inline CONTRIBUTING.md]: Plus 1

File: CONTRIBUTING.md (hunk 1 above)
Comment (robbieaverill) [inline CONTRIBUTING.md]: Do you want me to change this @Firesphere?

**Comment by robbieaverill** [comment]
Bump

File: README.md [hunk 3]
```diff
@@ -8,18 +8,32 @@
...
-This module generates @property, @method and @mixin tags for DataObjects, Page_Controllers and (Data)Extensions, so ide's like PHPStorm recognize the database and relations that are set in the $db, $has_one, $has_many and $many_many arrays.
+This module generates `@property`, `@method` and `@mixin` tags for sub classes of `DataObject`, `Page_Controller` and `Extension`/`DataExtension` so that IDEs like PHPStorm can recognize the database fields and relations that are set in `$db`, `$has_one`, `$has_many` and `$many_many` arrays.
 
-The docblocks can be generated/updated with each dev/build and with a DataObjectAnnotatorTask per module or classname.
+The docblocks can be generated/updated with each `dev/build` and with a `DataObjectAnnotatorTask` per module or classname.
 
-# Owners
+## Requirements
+
+* PHP 5.3.3+
```
Comment (Firesphere) [inline README.md]: 5.6 for master?

File: docs/en/Installation.md [hunk 4]
```diff
@@ -1,40 +1,49 @@
...
     environment: 'dev'
 ---
 DataObjectAnnotator:
-    enabled_modules:
-      - mysite
-      - otherfolderinsiteroot
-````
+  enabled_modules:
+    - mysite
+    - otherfolderinsiteroot
```
Comment (Firesphere) [inline docs/en/Installation.md]: This should be commented, together with a comment saying "for example". It's bad practice to enable something, even with a name like this.

File: docs/en/Installation.md [hunk 5]
```diff
@@ -1,40 +1,49 @@
...
 
-When enabled IdeAnnotator generates the docblocks on dev/build for mysite only.
+When enabled, IdeAnnotator generates the docblocks on `dev/build` for the `/mysite` folder only.
 
-You can add extra module folders with the following config setting :
+You can add extra module folders with the following config setting:
 
 ```php
 Config::inst()->update('DataObjectAnnotator', 'enabled_modules', array('mysite', 'otherfolderinsiteroot'));
 ```
```
Comment (Firesphere) [inline docs/en/Installation.md]: I'd like to get rid of the PHP config settings if possible for SS4

File: docs/en/Installation.md [hunk 6]
```diff
@@ -1,40 +1,49 @@
...
-##Config
-This module is disabled by default and I recommend to only enable this module in your local development environment, since this module changes the file content of the Dataobject and DataExtension classes.
+This module is disabled by default and it is recommended to only enable this module in your local development environment, since this module changes the file content of the `DataObject` and `DataExtension` classes.
 
-You can do this, by using something like this in your mysite/_config.php :
+You can do this by using something like this in your `mysite/_config.php`:
 
 ```php
-if($_SERVER['HTTP_HOST'] == 'mysite.local.dev') {
+if ($_SERVER['HTTP_HOST'] == 'mysite.local.dev') {
```
Comment (Firesphere) [inline docs/en/Installation.md]: Could we remove this too, replacing it with a `environment.only: dev` yml config? I want to steer away from php stuff when it comes to configs

File: README.md [hunk 7]
```diff
@@ -61,18 +75,13 @@ class NewsItem extends DataObject
...
-I tried to add complete UnitTests, but I can't garantuee every situation is covered.
+For further information please see the [user guide documentation](docs/en/Index.md).
 
+Please see [the contributing guide](CONTRIBUTING.md) for more information on how to contribute to this project.
 
+## Caution
 
+This module changes the content of your files and currently there is no backup functionality. PHPStorm has a Local history for files and of course you have your code version controlled... Unit tests are in place, but it can't be guaranteed that every situation is covered.
 
+Please use carefully, and only install in development environments.
```
Comment (Firesphere) [inline README.md]: 👍

File: docs/en/Installation.md [hunk 8]
```diff
@@ -1,40 +1,49 @@
...
 ---
 DataObjectAnnotator:
-    enabled_modules:
-      - mysite
-      - otherfolderinsiteroot
-````
+  enabled_modules:
+    - mysite
+    - otherfolderinsiteroot
+```
```
Comment (Firesphere) [inline docs/en/Installation.md]: Could you add a ## Known Issues

Currently, on SS4, extensions are not generated correctly, I'm working on that bit.

File: README.md (hunk 3 above)
Comment (robbieaverill) [inline README.md]: Master is still SS3 compatible, no?

**Comment by robbieaverill** [comment]
@Firesphere I don't have much time to address your other comments at the moment - this was just a quick whip over to fix formatting etc

**Pull request closed without merging**